token/cli/src/main.rs

```diff
@@ -7425,4 +7453,43 @@ mod tests {
                 .unwrap()
         );
     }
+
+    #[tokio::test]
+    #[serial]
+    async fn metadata_pointer() {
+        let (test_validator, payer) = new_validator_for_test().await;
+        let config =
+            test_config_with_default_signer(&test_validator, &payer, &spl_token_2022::id());
+        let program_id = &spl_token_2022::id();
+        let token_path = NamedTempFile::new().unwrap();
```

Comment:
Is this needed? Only tests that need the file to be written somewhere use it. Otherwise, the CLI will generate a new keypair, and you can pull it out from `value["commandOutput"]["address"]`, as you're already doing